test/mocha/gilded-rose.spec.ts

```diff
@@ -1,92 +1,170 @@
+  Brie = 'Aged Brie',
+  Pass = 'Backstage passes to a TAFKAL80ETC concert',
+  Conjured = 'Conjured',
+  Sulfuras = 'Sulfuras, Hand of Ragnaros',
+  Basic = 'item'
+}
 
-    expect(items[0].quality).to.equal(0);
-  });
-  it('aged brie should increse in quality by 1 if sellin > 0', () => {
-    const glidedRose = new GildedRose([new Item('Aged Brie', 1, 10)]);
-
-    const items = glidedRose.updateQuality();
-
-    expect(items[0].quality).to.equal(11);
-  });
-  it('aged brie should increse in quality by 2 if sellin <= 0', () => {
-    const glidedRose = new GildedRose([new Item('Aged Brie', 0, 10)]);
-
-    const items = glidedRose.updateQuality();
-
-    expect(items[0].quality).to.equal(12);
-  });
-  it('quality should not be above 50', () => {
-    const glidedRose = new GildedRose([new Item('Aged Brie', 0, 50)]);
-
-    const items = glidedRose.updateQuality();
-
-    expect(items[0].quality).to.equal(50);
-  });
-  it('sulfuras should not decrese in quality or ever be sold', () => {
-    const glidedRose = new GildedRose([new Item('Sulfuras, Hand of Ragnaros', 1, 20)]);
-
-    const items = glidedRose.updateQuality();
-
-    expect(items[0].quality).to.equal(20);
-    expect(items[0].sellIn).to.equal(1);
-  });
-  it('backstage pass should increase in quality by 1 if sellIn > 10', () => {
-    const glidedRose = new GildedRose([new Item('Backstage passes to a TAFKAL80ETC concert', 11, 20)]);
-
-    const items = glidedRose.updateQuality();
-
-    expect(items[0].quality).to.equal(21);
+describe('Gilded Rose', () => {
+  let testedItem: string = '';
+  describe('Non-special item', () => {
+    before(() => {
+      testedItem = Items.Basic;
+    });
+
+    it('should degrade quality by one if sellin > 0', () => {
+      //Given
+      const glidedRose = new GildedRose([new Item(testedItem, 1, 5)]);
+      //When
+      const items = glidedRose.updateQuality();
+      //Then
+      expect(items[0].quality).to.equal(4);
+    });
+
+    it('should degrade quality twice as fast if sellin <= 0', () => {
+      //Given
+      const glidedRose = new GildedRose([new Item(testedItem, 0, 5)]);
+      //When
+      const items = glidedRose.updateQuality();
+      //Then
+      expect(items[0].quality).to.equal(3);
+    });
+
+    it('should not degrade anymore the quality if zero', () => {
+      //Given
+      const glidedRose = new GildedRose([new Item(testedItem, 0, 0)]);
+      //When
+      const items = glidedRose.updateQuality();
+      //Then
+      expect(items[0].quality).to.equal(0);
+    });
   });
-  it('backstage pass should increase in quality by 2 if sellIn <= 10', () => {
-    const glidedRose = new GildedRose([new Item('Backstage passes to a TAFKAL80ETC concert', 10, 20)]);
 
-    const items = glidedRose.updateQuality();
-
-    expect(items[0].quality).to.equal(22);
+  describe('Brie', () => {
+    before(() => {
+      testedItem = Items.Brie;
+    });
+
+    it('aged brie should increse in quality by 1 if sellin > 0', () => {
```

Comment:
we could remove the item type (aged brie in this case) from the names of the tests